### DISCUSSION TITLE
quering post by id

### DISCUSSION THREAD

**Discussion by ashishch-98** [gatsbyjs/gatsby#28587] [Help]
Gatsby is awesome to use and I am using it dynamically for my PWA. Now I have connected my PWA with firebase to use thier products. I am adding  a feature by which user can add the post to read later list. So I am adding the markdown blog id to users collection in firebase. Now on a client route for ex: _'/user/read-later'_  I want to show the user the list of all the blogs which he added to the collection to read later. 

So to achieve this I want to pass that id to the query and then get the post. So basically it will loop for all the ids stored by user and return them as a list with the properties of like title, author, excerpt etc for the respective id. I tried passing the id as a variable to static query in the component by then got to know that variables cant be passed to static queries. 

**Question:** How can I get the posts by passing their ids to the query?

**Comment by vladar** [comment]
Can you share some code that you've already tried and that didn't work? Maybe someone will be able to help. It's hard to help just by reading this description.
